test/integration/redirects_test.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require("test_helper")
+
+class RedirectsTest < IntegrationTestCase
```

Comment:
@mo-nathan:
1. I assume you prefer 1 get one per method.
2. Have I overdone the number of tests: 1 test  / old action / request method? (Exception: destroy)

Reply:
PS:
I wonder if a better approach is:
- Run `rails routes`
- Check the output for the expected routes.

That would test only whether `routes.rb` generates the expected redirects, which is what I originally set out to do. It's not dependent on what Rails does with one of the redirected requests. (That can be a little funky, e.g., PATCH|PUT `destroy` vs POST 'destroy`.) No http requests are needed.
On the other hand, it's dependent on how Rails formats the routes.

Reply:
I've now implemented that proposal as `test/unit/redirects_test.rb`.
Bonus: on my old MacBook Pro, the unit test is faster by ~20% (4 seconds).

Reply:
Is 4 secs/test or 4 secs overall?

Using the output of a Rails command that is intended for human readability seems a bit fragile to me.  For example, I know when I run `rails routes` at work, I get one line per verb and the Prefix is not repeated (e.g., https://backend.turing.io/module2/lessons/archive/advanced_routing_rails).  I haven't figured out why we aren't seeing that, but it seems equally likely that it's because we're using a more recent release and we're using a more recent release at work.  In any case, I think the point remains.  It also just feels weird to me to be testing the results of a shell command (although I think we're doing that in other places).

Reply:
Good point about fragility. I'll stick with the integration test.
(And it's 4 sec. overall. Probably less on other machines.)